configure.ac

```diff
@@ -255,24 +255,36 @@ AX_APPEND_COMPILE_FLAGS([-Wextra -Werror -pedantic])
 # ---------------------------------------------------------------
 # Enable compile-time defense
 # ---------------------------------------------------------------
-# Fortify source
-# Enabling optimization implies _FORTIFY_SOURCE on some platforms.
-# Explicitly redefine to _FORTIFY_SOURCE=2 to make sure we have the
-# desired fortification level.
-AX_APPEND_FLAG([-U_FORTIFY_SOURCE], [CPPFLAGS])
-AX_APPEND_FLAG([-D_FORTIFY_SOURCE=2], [CPPFLAGS])
+
+AC_ARG_ENABLE(stack-protection,
+	[AS_HELP_STRING([--enable-stack-protection],
+		[ Enable compiler stack protection.
+		FORTIFY_SOUCE=2 and -fstack-protector-strong]
+	)]
```

Comment:
I'd rather leave these security features enabled by default, and instead have the user explicitly disable them through a `--disable-stack-protection` `configure` script command line option.  That way we won't alter the long-standing default build configuration.  Please make the `AC_ARG_ENABLE()` _action-if-not-given_ argument be `enable_stack_protection=yes`, and adjust the help string accordingly.  See the [AC_ARG_WITH()](https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/External-Software.html#External-Software) documentation in the GNU Autoconf for a similar "enabled by default" case.

Reply:
done